src/main/scala/org/hammerlab/guacamole/distributed/RegionsByContig.scala

```diff
@@ -20,22 +20,30 @@ import org.hammerlab.guacamole.reference.ReferenceRegion
  * @param regionIterator regions, sorted by contig and start locus.
  */
 class RegionsByContig[R <: ReferenceRegion](regionIterator: Iterator[R]) {
+
   private val buffered = regionIterator.buffered
   private var seenContigs = List.empty[String]
-  private var prevIterator: Option[SingleContigRegionIterator[R]] = None
+  private var curIterator: SingleContigRegionIterator[R] = _
+
   def next(contig: String): Iterator[R] = {
-    // We must first march the previous iterator we returned to the end.
-    while (prevIterator.exists(_.hasNext)) prevIterator.get.next()
```

Comment:
no semantic change intended; I lost a bit in my rebasing here which caused the last travis failure. should be fixed now.

I've gone back and forth on using `null`s vs. `None`s in various Iterator implementations; I used to be big on `Option`s for these cases but a lot of `Iterator`s are not written that way and it seems cleaner not to in a lot of cases.

For example, here the `null` is only there for the very first time the Iterator runs; the variable is logically not optional in the steady state, but is more like lazily-initialized. Anyway, just a gut thing, if you don't like it I can revert it. My initial idea here was to call it `curIterator` instead of `prevIterator` which seems clearer.